marlowe-playground-client/src/SimulationPage/View.purs

```diff
@@ -488,23 +493,11 @@ inputItem _ state person (MoveToSlot slot) =
 
   boundsError = "The slot must be more than the current slot " <> (state ^. (_currentMarloweState <<< _executionState <<< _SimulationRunning <<< _slot <<< to show))
 
-marloweActionInput :: forall p a action. Show a => (BigInteger -> action) -> a -> HTML p action
-marloweActionInput f current =
-  input
-    [ type_ InputNumber
-    , placeholder "BigInteger"
-    , class_ $ ClassName "action-input"
-    , value $ show current
-    , onValueChange
-        $ ( \x ->
-              Just
-                $ f
-                    ( case fromString x of
-                        Just y -> y
-                        Nothing -> fromInt 0
-                    )
-          )
-    ]
+marloweCurrencyInput :: forall p action. (BigInteger -> action) -> String -> Int -> BigInteger -> HTML p action
+marloweCurrencyInput f currencyLabel numDecimals current = currencyInput [ "mx-2", "flex-grow", "flex-shrink-0" ] current currencyLabel numDecimals (Just <<< f <<< fromMaybe zero)
```

Comment:
why are you passing flex-grow and shrink0 from the parent and not defined in the currencyInput?

Reply:
My thought was that the child cannot know whether the parent is flex, so I thought it would be wrong to say `flex-grow` and `flex-shrink-0`, if the parent turns out not to be flex

Reply:
It cannot know and it shouldn't know or care... I think is better to assume that the input will occupy all the space available (`currencyInput` should have "w-full") and it's up to the parent (whoever includes `marloweCurrencyInput`, which by the way also doesn't know if the parent has flex) to limit the width.

Reply:
I can put `w-full` if you prefer, but I don't understand, if the parent means to use the child as part of flex, isn't it necessary to specify in the child the kind of flex item it wants it to be? That would be done through a parameter, right?

Reply:
I think there are two sensitive approaches:
* One is to make this `w-full` and then in the parent limit the size (most likely adding a div/span wrapper to this component)
* The other is to have the parameter as you mention, but this probably should not be the place to set those parameters, rather in the place that you are using it (will put a comment there)